.github/workflows/test.yml

```diff
@@ -0,0 +1,52 @@
+name: KBase Auth Client Tests
+
+on:
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
+    - ready_for_review
+  push:
+    # run workflow when merging to main or develop
+    branches:
+      - main
+      - develop
+
+jobs:
+
+  auth_client_tests:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
+          - python-version: "3.12"
+
+    steps:
+    
+    - name: Repo checkout
+      uses: actions/checkout@v4
+    
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
+
+    - name: Install dependencies
+      shell: bash
+      run: |
+        export UV_PROJECT_ENVIRONMENT="${pythonLocation}"
+        uv sync --locked
```

Comment:
uv can also manage your python installation if you have committed the `.python-version` file

Reply:
Right, but the current way means we can matrix test multiple versions of python if we want to, which I'm guessing we'll probably want to